sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureBlobFileAttributes.java

```diff
@@ -319,7 +426,11 @@ public boolean isOther() {
      */
     @Override
     public long size() {
-        return properties.getBlobSize();
+        if (!this.isVirtualDirectory) {
+            return properties.getBlobSize();
+        } else {
+            return 0;
+        }
```

Comment:
As a minor stylistic comment (and as such, I'm not expecting anything from saying this), one pattern I've favoured over the years is to reverse this code approach, to reduce indentation and 'fall out' of the method early, rather than late. In this case, I would have considered doing it as such:

```java
public long size() {
    if (this.isVirtualDirectory) {
        return 0;
    }
    return properties.getBlobSize();
}
```

In some projects, we even had a code style expectation that such simple return statements didn't need to have braces, so we would have made it simpler and on one line, as below:

```java
public long size() {
    if (this.isVirtualDirectory) return 0;
    return properties.getBlobSize();
}
```

Of course, at this point we might as well go the whole way and just do a single-line ternary expression:

```java
public long size() {
    return this.isVirtualDirectory ? 0 : properties.getBlobSize();
}
```

Reply:
Good idea. I can turn these into ternary expressions. Much cleaner